### PR TITLE
sparse BERT + Distilation

### PR DESCRIPTION
**Pull request by spacemanidol**

### CODE REVIEW

**Comment by spacemanidol** [comment]
base code is up. Working on cleaning up the scripts for public usage and improved integration with huggingface

**Comment by spacemanidol** [comment]
Code cleaned up for public usage. Awaiting results of distillation with pruning and layer dropping. All other components are ready for review @bfineran @markurtz.

**Review by jeanniefinks — CHANGES_REQUESTED** [review]
This file needs several editorial updates.
e.g., demostrate or absed or Distilation

Consider taking another pass through it and happy to be added as a reviewer!

**Comment by spacemanidol** [comment]
@jeanniefinks please go ahead and call our any changes required. At this point, I am just awaiting results on experiments.

**Review by markurtz — DISMISSED** [review]

File: integrations/transformers/run_qa.py [hunk 1]
```diff
@@ -435,6 +441,7 @@ def dropLayers(model, layers_to_keep):
 
 def main():
     ### Dataset processing classes in main due to hugging face custom dataset map
+    wandb.init(project='BERT-QA-SPARSEML', entity='spacemanidol') # REMOVE before pushing code
```
Comment (bfineran) [inline integrations/transformers/run_qa.py:444]: make sure to remove

File: integrations/transformers/run_qa.py [hunk 2]
```diff
@@ -711,6 +719,9 @@ def prepare_validation_features(examples):
     optim = load_optimizer(model, TrainingArguments)
     steps_per_epoch = math.ceil(len(datasets["train"]) / (training_args.per_device_train_batch_size*training_args._n_gpu))
     manager = ScheduledModifierManager.from_yaml(data_args.nm_prune_config)
+    print(training_args.num_train_epochs)
```
Comment (bfineran) [inline integrations/transformers/run_qa.py:722]: nit: can we use the `logger` here?

File: integrations/transformers/run_qa.py [hunk 3]
```diff
@@ -711,6 +719,9 @@ def prepare_validation_features(examples):
...
+    training_args.num_train_epochs = float(manager.modifiers[0].end_epoch)
+    print(training_args.num_train_epochs)
```
Comment (bfineran) [inline integrations/transformers/run_qa.py:724]: nit: can we use the logger here?

**Review by bfineran — CHANGES_REQUESTED** [review]
LGTM outside of nits and the extra wandb line. Would it be possible in the future to compress the recipes to just use a single `GMPruningModifier` if the params are the same across all layers?

File: integrations/transformers/run_qa.py (hunk 3 above)
Comment (spacemanidol) [reply integrations/transformers/run_qa.py]: these are unneeded so I dropped them

File: integrations/transformers/run_qa.py (hunk 2 above)
Comment (spacemanidol) [reply integrations/transformers/run_qa.py]: unneeded so removed

File: integrations/transformers/run_qa.py (hunk 1 above)
Comment (spacemanidol) [reply integrations/transformers/run_qa.py]: wandb gone.

**Review by bfineran — DISMISSED** [review]

**Review by markurtz — APPROVED** [review]

**Pull request merged**